pandas/tests/arrays/categorical/test_api.py

```diff
@@ -337,12 +338,31 @@ def test_remove_categories(self):
         # inplace == True
         res = cat.remove_categories("c", inplace=True)
         tm.assert_categorical_equal(cat, new)
+
+        # for removing duplicates
+        cat = Categorical(["a", "b", "c", "a"], ordered=True)
+        old = cat.copy()
+        new = Categorical([np.nan, "b", "c", np.nan], categories=["b", "c"], ordered=True)
+
+        res = cat.remove_categories("a")
+        tm.assert_categorical_equal(cat, old)
+        tm.assert_categorical_equal(res, new)
+
+        res = cat.remove_categories(["a"])
+        tm.assert_categorical_equal(cat, old)
+        tm.assert_categorical_equal(res, new)
+
         assert res is None
 
         # removal is not in categories
-        with pytest.raises(ValueError):
+        msg = re.escape("removals must all be in old categories: ['c']")
+        with pytest.raises(ValueError, match=msg):
             cat.remove_categories(["c"])
 
+        msg = re.escape("removals must all be in old categories: [np.nan]")
```

Comment:
i think we should have 4 tests. that all produce the same error message so only need to assign to msg once.